raster_api/runtime/src/app.py

```diff
@@ -1,124 +1,114 @@
+
 
 if settings.debug:
     optional_headers = [OptionalHeader.server_timing, OptionalHeader.x_assets]
 else:
     optional_headers = []
 
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    """FastAPI Lifespan."""
+    # Create Connection Pool
+    await connect_to_db(app, settings=settings.load_postgres_settings())
+    yield
+    # Close the Connection Pool
+    await close_db_connection(app)
+
+
 path_prefix = settings.path_prefix
 app = FastAPI(
     title=settings.name,
     version=veda_raster_version,
     openapi_url=f"{path_prefix}/openapi.json",
     docs_url=f"{path_prefix}/docs",
+    lifespan=lifespan,
 )
 
 # router to be applied to all titiler route factories (improves logs with FastAPI context)
 router = APIRouter(route_class=LoggerRouteHandler)
 add_exception_handlers(app, DEFAULT_STATUS_CODES)
 add_exception_handlers(app, MOSAIC_STATUS_CODES)
 
-
-# Custom PgSTAC mosaic tiler
+###############################################################################
+# /mosaic - PgSTAC Mosaic titiler endpoint
+###############################################################################
 mosaic = MosaicTilerFactory(
     router_prefix=f"{path_prefix}/mosaic",
     add_mosaic_list=settings.enable_mosaic_search,
     optional_headers=optional_headers,
     environment_dependency=settings.get_gdal_config,
-    dataset_dependency=DatasetParams,
+    post_process=PostProcessParams,
     router=APIRouter(route_class=LoggerRouteHandler),
 )
 app.include_router(mosaic.router, prefix=f"{path_prefix}/mosaic", tags=["Mosaic"])
+# TODO
+# prefix will be replaced by `/mosaics/{search_id}` in titiler-pgstac 0.9.0
 
-# Custom STAC titiler endpoint (not added to the openapi docs)
+###############################################################################
+# /stac - Custom STAC titiler endpoint
+###############################################################################
 stac = MultiBaseTilerFactory(
     reader=PgSTACReader,
     path_dependency=ItemPathParams,
     optional_headers=optional_headers,
     router_prefix=f"{path_prefix}/stac",
     environment_dependency=settings.get_gdal_config,
     router=APIRouter(route_class=LoggerRouteHandler),
+    extensions=[
+        stacViewerExtension(),
+    ],
 )
 app.include_router(stac.router, tags=["Items"], prefix=f"{path_prefix}/stac")
+# TODO
+# in titiler-pgstac we replaced the prefix to `/collections/{collection_id}/items/{item_id}`
 
+###############################################################################
+# /cog - External Cloud Optimized GeoTIFF endpoints
+###############################################################################
 cog = TilerFactory(
     router_prefix=f"{path_prefix}/cog",
     optional_headers=optional_headers,
     environment_dependency=settings.get_gdal_config,
     router=APIRouter(route_class=LoggerRouteHandler),
+    extensions=[
+        cogValidateExtension(),
+        cogViewerExtension(),
+    ],
```

Comment:
Using the official titiler extensions for the `/validate` and `/viewer` endpoints